src/python/WMCore/Storage/StageOutImpl.py

```diff
@@ -224,10 +235,15 @@ def __call__(self, protocol, inputPFN, targetPFN, options=None, checksums=None):
                 logging.error(msg)
                 if retryCount == self.numRetries:
                     #  //
-                    # // last retry, propagate exception
+                    # // last retry, propagate the information outside of the for loop
                     # //
-                    raise ex
+                    stageOutEx = ex
                 time.sleep(self.retryPause)
 
+        if stageOutEx is not None:
```

Comment:
This idea is looking good. Please surround it with some relevant log records though.